test/integration/utils.go

```diff
@@ -0,0 +1,350 @@
+	return []corev1.VolumeMount{
+		{
+			Name:      "log",
+			MountPath: "/var/log",
+		},
+		{
+			Name:      "cns-state",
+			MountPath: "/var/lib/azure-network",
+		},
+		{
+			Name:      "cns-config",
+			MountPath: "/etc/azure-cns",
+		},
+		{
+			Name:      "cni-bin",
+			MountPath: "/opt/cni/bin",
+		},
+		{
+			Name:      "azure-vnet",
+			MountPath: "/var/run/azure-vnet",
+		},
+		{
+			Name:      "cni-lock",
+			MountPath: "/var/lock/azure-vnet",
+		},
+		{
+			Name:      "legacy-cni-state",
+			MountPath: "/var/run/azure-vnet.json",
+		},
+		{
+			Name:      "cni-conflist",
+			MountPath: "/etc/cni/net.d",
+		},
+	}
+}
+
+func loadCNSDaemonset(ctx context.Context, clientset *kubernetes.Clientset, cnsVersion, cniDropgzVersion string) (appsv1.DaemonSet, error) {
+	cns, err := k8sutils.MustParseDaemonSet(cnsDaemonSetPath)
+	if err != nil {
+		return appsv1.DaemonSet{}, errors.Wrapf(err, "failed to parse daemonset")
+	}
+
+	image, _ := k8sutils.ParseImageString(cns.Spec.Template.Spec.Containers[0].Image)
+	cns.Spec.Template.Spec.Containers[0].Image = k8sutils.GetImageString(image, cnsVersion)
+
+	log.Printf("Checking environment scenario")
+	cns = loadDropgzImage(cns, cniDropgzVersion)
+	cns, err = loadAzureVnet(ctx, clientset, cns)
+	if err != nil {
+		return appsv1.DaemonSet{}, errors.Wrap(err, "failed to load azure vnet")
+	}
+	cns, err = loadAzilium(ctx, clientset, cns)
+	if err != nil {
+		return appsv1.DaemonSet{}, errors.Wrap(err, "failed to load azilium")
+	}
+	cns, err = loadCiliumOverlay(ctx, clientset, cns)
+	if err != nil {
+		return appsv1.DaemonSet{}, errors.Wrap(err, "failed to load cilium overlay")
+	}
+	cns, err = loadAzureCNIOverlay(ctx, clientset, cns)
+	if err != nil {
+		return appsv1.DaemonSet{}, errors.Wrap(err, "failed to load azure cni overlay")
+	}
+	cns, err = loadDualstackOverlay(ctx, clientset, cns)
+	if err != nil {
+		return appsv1.DaemonSet{}, errors.Wrap(err, "failed to load dualstack overlay")
+	}
+
+	cnsDaemonsetClient := clientset.AppsV1().DaemonSets(cns.Namespace)
+
+	log.Printf("Installing CNS with image %s", cns.Spec.Template.Spec.Containers[0].Image)
+
+	// setup common RBAC, ClusteerRole, ClusterRoleBinding, ServiceAccount
+	if _, err := k8sutils.MustSetUpClusterRBAC(ctx, clientset, cnsClusterRolePath, cnsClusterRoleBindingPath, cnsServiceAccountPath); err != nil {
+		return appsv1.DaemonSet{}, errors.Wrap(err, "failed to setup common RBAC, ClusteerRole, ClusterRoleBinding and ServiceAccount")
+	}
+
+	// setup RBAC, Role, RoleBinding
+	if err := k8sutils.MustSetUpRBAC(ctx, clientset, cnsRolePath, cnsRoleBindingPath); err != nil {
+		return appsv1.DaemonSet{}, errors.Wrap(err, "failed to setup RBAC, Role and RoleBinding")
+	}
+
+	if err := k8sutils.MustCreateDaemonset(ctx, cnsDaemonsetClient, cns); err != nil {
+		return appsv1.DaemonSet{}, errors.Wrap(err, "failed to create daemonset")
+	}
+
+	if err := k8sutils.WaitForPodsRunning(ctx, clientset, cns.Namespace, cnsLabelSelector); err != nil {
```

Comment:
https://github.com/Azure/azure-container-networking/pull/2187#discussion_r1329412518 Can be used here

Reply:
Done